src/librustc/ty/subst.rs

```diff
@@ -179,6 +179,30 @@ impl<'a, 'gcx, 'tcx> Substs<'tcx> {
         })
     }
 
+    pub fn bound_vars_for_item(
```

Comment:
It'd be nice to have a comment here.

---

Returns a substitution vector containing "identity references" suitable for converting from the compiler's parameter representation to the bound type representation used in Chalk. The Nth parameter is substituted to a bound variable of index N, pointing at the innermost binder.

Reply:
I see you added this later :)